service/service.go

```diff
@@ -0,0 +1,190 @@
+package service
+
+import (
+	"context"
+	"github.com/ONSdigital/dp-search-api/api"
+	"github.com/ONSdigital/dp-search-api/elasticsearch"
+	"github.com/ONSdigital/dp-search-api/query"
+	"github.com/ONSdigital/dp-search-api/transformer"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"os"
+
+	esauth "github.com/ONSdigital/dp-elasticsearch/v2/awsauth"
+	elastic "github.com/ONSdigital/dp-elasticsearch/v2/elasticsearch"
+	dphttp "github.com/ONSdigital/dp-net/http"
+	"github.com/ONSdigital/dp-search-api/config"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+type Service struct {
+	config              *config.Configuration
+	healthCheck         HealthChecker
+	api                 *api.SearchAPI
+	router              *mux.Router
+	server              HTTPServer
+	serviceList         *ExternalServiceList
+	esSigner            *esauth.Signer
+	elasticHTTPClient   dphttp.Clienter
+	queryBuilder        api.QueryBuilder
+	elasticSearchClient elasticsearch.Client
+	transformer         transformer.Transformer
+}
+
+//New creates a new service
+func New(cfg *config.Configuration, serviceList *ExternalServiceList) *Service {
+	svc := &Service{
+		config:      cfg,
+		serviceList: serviceList,
+	}
+	return svc
+}
+
+// SetServer sets the http server for a service
+func (svc *Service) SetServer(server HTTPServer) {
+	svc.server = server
+}
+
+// SetHealthCheck sets the healthchecker for a service
+func (svc *Service) SetHealthCheck(healthCheck HealthChecker) {
+	svc.healthCheck = healthCheck
+}
+
+// SetQueryBuilder sets the queryBuilder for a service
+func (svc *Service) SetQueryBuilder(queryBuilder api.QueryBuilder) {
+	svc.queryBuilder = queryBuilder
+}
+
+// SetEsSigner sets the AWS signer for a service
+func (svc *Service) SetEsSigner(esSigner *esauth.Signer) {
+	svc.esSigner = esSigner
+}
+
+//SetElasticHTTPClient sets the HTTP client for a service
+func (svc *Service) SetElasticHTTPClient(elasticHTTPClient dphttp.Clienter) {
+	svc.elasticHTTPClient = elasticHTTPClient
+}
+
+//SetElasticSearchClient sets the new instance of elasticsearch for a service
+func (svc *Service) SetElasticSearchClient(elasticSearchClient elasticsearch.Client) {
+	svc.elasticSearchClient = elasticSearchClient
+}
+
+//SetTransformer sets the transformer for a service
+func (svc *Service) SetTransformer(transformer transformer.Transformer) {
+	svc.transformer = transformer
+}
+
+//Run the service
+func (svc *Service) Run(ctx context.Context, buildTime, gitCommit, version string, svcErrors chan error) (err error) {
+	elasticHTTPClient := dphttp.NewClient()
+	//Get HealthCheck
+	svc.healthCheck, err = svc.serviceList.GetHealthCheck(svc.config, buildTime, gitCommit, version)
+	if err != nil {
+		log.Fatal(ctx, "could not instantiate healthcheck", err)
+		return err
+	}
+	if err := svc.registerCheckers(ctx, elasticHTTPClient); err != nil {
```

Comment:
Worth adding a new line before this one - for ease of reading separate code blocks/statements